internal/service/flink/flink_application_version.go

```diff
@@ -0,0 +1,225 @@
+		Description: "Application sources",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"create_table": {
+					Type:        schema.TypeString,
+					Required:    true,
+					ForceNew:    true,
+					Description: "The CREATE TABLE statement",
+				},
+				"integration_id": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "The integration ID",
+				},
+			},
+		},
+	},
+	"statement": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Job SQL statement",
+	},
+
+	// Computed fields.
+	"application_version_id": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Application version ID",
+	},
+	"created_at": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Application version creation time",
+	},
+	"created_by": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Application version creator",
+	},
+	"version": {
+		Type:        schema.TypeInt,
+		Computed:    true,
+		Description: "Application version number",
+	},
+}
+
+// ResourceFlinkApplicationVersion returns the Flink Application Version resource schema.
+func ResourceFlinkApplicationVersion() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Flink Application Version resource allows the creation and management of Aiven Flink Application Versions.",
+		CreateContext: resourceFlinkApplicationVersionCreate,
+		ReadContext:   resourceFlinkApplicationVersionRead,
+		DeleteContext: resourceFlinkApplicationVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: aivenFlinkApplicationVersionSchema,
+	}
+}
+
+// resourceFlinkApplicationVersionCreate is the create function for the Flink Application Version resource.
+func resourceFlinkApplicationVersionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	serviceName := d.Get("service_name").(string)
+	applicationID := d.Get("application_id").(string)
+
+	r, err := client.FlinkApplicationVersions.Create(project, serviceName, applicationID, aiven.GenericFlinkApplicationVersionRequest{
+		Statement: d.Get("statement").(string),
+		Sources:   expandFlinkApplicationVersionSourcesOrSinks(d.Get("sources").(*schema.Set).List()),
+		Sinks:     expandFlinkApplicationVersionSourcesOrSinks(d.Get("sinks").(*schema.Set).List()),
+	})
+	if err != nil {
+		return diag.Errorf("cannot create Flink Application Version: %+v - %v", expandFlinkApplicationVersionSourcesOrSinks(d.Get("sources").(*schema.Set).List()), err)
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, serviceName, applicationID, r.ID))
+
+	return resourceFlinkApplicationVersionRead(ctx, d, m)
+}
+
+func expandFlinkApplicationVersionSourcesOrSinks(sources []interface{}) []aiven.FlinkApplicationVersionRelation {
```

Comment:
nit: could we have all functions commented please?

Reply:
ok, no problem added more comments even to the private functions